Tests/MongoSwiftTests/EventLoopBoundMongoClientTests.swift

```diff
@@ -0,0 +1,149 @@
+@testable import MongoSwift
+import Nimble
+import NIO
+import TestsCommon
+
+final class EventLoopBoundMongoClientTests: MongoSwiftTestCase {
```

Comment:
throughout these tests we probably want to `wait()` on any future we create before returning, even if we don't care about the result or whether it actually executes. 

otherwise, I think we may run into race conditions around the futures completing and the client being closed by `withTestClient`, which may cause some sporadic errors.